edb/server/main.py

```diff
@@ -468,7 +475,8 @@ def bump_rlimit_nofile() -> None:
     click.option(
         '--runstate-dir', type=PathPath(), default=None,
         help='directory where UNIX sockets will be created '
-             '("/run" on Linux by default)'),
+             '(data-dir or {!r} by default)'
```

Comment:
This will print `or 'None'` for non-production builds, which isn't very nice.  I'd rewrite `get_runstate_dir()` as:

```python

def _get_runstate_dir_default():
    try:
        return buildmeta.get_build_metadata_value("RUNSTATE_DIR")
    except buildmeta.MetadataError:
        return '<data-dir>`
```

Reply:
and then:

```python
help = f'directory where UNIX sockets and other temporary '
       f'runtime files will be placed ({_get_runstate_dir_default()} by default)'
```
```

Reply:
As far as I understand the `--runstate-dir` is required if no `RUNSTATE_DIR` present in buildmeta and not a `--devmode`.  So I'm not sure how far we want to go here with explanations.

Reply:
> As far as I understand the --runstate-dir is required if no RUNSTATE_DIR present in buildmeta and not a --devmode.

Buildmeta and `devmode` are complementary, either is supposed to be present.

Reply:
Okay. Updated.